googleapiclient/http.py

```diff
@@ -190,11 +187,16 @@ def _retry_request(
             exception = connection_error
         except socket.error as socket_error:
             # errno's contents differ by platform, so we have to match by name.
+            # Some of these same errors may have been caught above, e.g. ECONNRESET *should* be
+            # raised as a ConnectionError, but some libraries will raise it as a socket.error
+            # with an errno corresponding to ECONNRESET
             if socket.errno.errorcode.get(socket_error.errno) not in {
                 "WSAETIMEDOUT",
                 "ETIMEDOUT",
                 "EPIPE",
                 "ECONNABORTED",
+                "ECONNREFUSED",
```

Comment:
@dossett Thanks for the fix! The test [TestHttpRequest.test_no_retry_connection_errors](https://github.com/googleapis/google-api-python-client/blob/master/tests/test_http.py#L937) is failing. I think it's because the test is [expecting OSError to be raised ](https://github.com/googleapis/google-api-python-client/blob/master/tests/test_http.py#L174) when the `socket.errno` is `ECONNREFUSED`. The test passes if I remove `ECONNREFUSED` from this list. Please can you take a look?

Reply:
Thanks for pointing that out @parthea.  I made a small change to `test_http` to have it use `EHOSTUNREACH` for errors that shouldn't be retried.  That's just the error that caught my eye when I looked through a list of error codes, but it does seem to work ok.